ga/developer/kernel/Dockerfile

```diff
@@ -48,5 +48,14 @@ RUN /opt/ibm/wlp/bin/server create \
 COPY docker-server /opt/ibm/docker/
 EXPOSE 9080 9443
 
+# These instructions will be run in images built FROM this image
+ONBUILD COPY server.xml /config/
+ONBUILD ARG REPOSITORIES_PROPERTIES=""
+ONBUILD RUN if [ ! -z $REPOSITORIES_PROPERTIES ]; then mkdir /opt/ibm/wlp/etc/ \
+  && echo $REPOSITORIES_PROPERTIES > /opt/ibm/wlp/etc/repositories.properties; fi \
+  && installUtility install --acceptLicense defaultServer \
+  && if [ ! -z $REPOSITORIES_PROPERTIES ]; then rm /opt/ibm/wlp/etc/repositories.properties; fi \
+  && server start && server stop && rm -rf /output/logs
```

Comment:
I didn't find this immediately intuitive. I was wondering why javaee-7.0 and microProfile-1.3 wasn't in the install list from the webProfile, microProfile and javaee images based on this. I think it would be a better idea to put them in the child ones and skip the install defaultServer. I think it is more obvious about what is going on. This just makes me question why we are doing the install in the child images. I know this is to make sure they don't disappear, but this is something that is in the 'smart, but confusing for the idiot that is me in a months time trying to understand this again' category.

For the javaee-7.0 image you should remove the message store files which are stored in /output/messaging. They are something like 100Mb of dead space that isn't needed. It'll be created on docker run and is pretty fast and efficient.

The command doesn't remove the generated SSL certificate and LTPA keys which means the whole world of docker is using the same keys for this which is a security issue. These can be removed by deleting the content of /output/resources/security/

Reply:
Thanks for the feedback Alasdair.   The main goal of the PR was to automatic have the start/stop code running for all kernel-derived images..but the issue is that we need to have a server.xml and its features installed for the start/stop priming to take effect - which led to copying server.xml and running installUtility against it prior to priming.  

I agree though, that this may not be so intuitive at first glance.  I am looking into options to make this a bit clearer.  

Great points about the other cleanup needed - definitely will address that.